memstore/src/lib.rs

```diff
@@ -233,16 +233,11 @@ impl RaftStorage<ClientRequest, ClientResponse> for MemStore {
         Ok(log.get(&log_index).cloned())
     }
 
-    async fn first_id_in_log(&self) -> Result<Option<LogId>, StorageError> {
+    async fn get_log_state(&self) -> Result<(Option<LogId>, Option<LogId>), StorageError> {
         let log = self.log.read().await;
         let first = log.iter().next().map(|(_, ent)| ent.log_id);
-        Ok(first)
-    }
-
-    async fn last_id_in_log(&self) -> Result<LogId, StorageError> {
-        let log = self.log.read().await;
-        let last = log.iter().last().map(|(_, ent)| ent.log_id).unwrap_or_default();
-        Ok(last)
+        let last = log.iter().last().map(|(_, ent)| ent.log_id);
```

Comment:
`last()` seems tend to traverse all of the elements.
```suggestion
        let last = log.iter().rev().next().map(|(_, ent)| ent.log_id);
```

Reply:
done